buildenv/jenkins/docker-slaves/x86/centos6.9/Dockerfile

```diff
@@ -315,6 +290,38 @@ RUN yum -y update \
   && cd .. \
   && rm -rf Python-3.7.3
 
+# Any commands that require gcc need to be run before this SHELL command and after the previous SHELL
+SHELL ["/bin/sh","-c"]
+
+# add user home/jenkins
+RUN useradd -ms /bin/bash ${USER} \
+  && mkdir /home/${USER}/.ssh/
+COPY authorized_keys /home/${USER}/.ssh/authorized_keys
+COPY known_hosts /home/${USER}/.ssh/known_hosts
+RUN chown -R ${USER}:${USER} /home/${USER} \
+  && chmod 644 /home/${USER}/.ssh/authorized_keys \
+  && chmod 644 /home/${USER}/.ssh/known_hosts
+
+# Install Freemaker for building OpenJ9. Used in bash ./configure --with-freemarker-jar=<path-to-freemaker-jar>
+RUN cd /home/${USER} \
+  && wget https://sourceforge.net/projects/freemarker/files/freemarker/2.3.8/freemarker-2.3.8.tar.gz/download -O freemarker.tgz \
+  && tar -xzf freemarker.tgz freemarker-2.3.8/lib/freemarker.jar --strip=2 \
+  && rm -f freemarker.tgz
+
+# Setup a reference repository cache for faster clones in the container
+RUN mkdir /home/${USER}/openjdk_cache \
+  && cd /home/${USER}/openjdk_cache \
+  && git init --bare \
+  && git remote add jdk8 https://github.com/ibmruntimes/openj9-openjdk-jdk8.git \
+  && git remote add jdk11 https://github.com/ibmruntimes/openj9-openjdk-jdk11.git \
+  && git remote add jdk12 https://github.com/ibmruntimes/openj9-openjdk-jdk12.git \
+  && git remote add jdk13 https://github.com/ibmruntimes/openj9-openjdk-jdk13.git \
```

Comment:
This is likely to run more quickly if `jdk13` is the first remote: that repo was subjected to `gc --aggressive` and so is much smaller than the others even though there's a great deal of commonality among the repos.